src/storage/src/client.rs

```diff
@@ -313,6 +277,200 @@ pub(crate) mod info {
+        self.request.bucket = v.into();
+        self
+    }
+
+    /// Sets the value of [object][control::model::ReadObjectRequest::object].
+    pub fn set_object<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
+        self.request.object = v.into();
+        self
+    }
+
+    /// Sets the value of [generation][control::model::ReadObjectRequest::generation].
+    pub fn set_generation<T: std::convert::Into<i64>>(mut self, v: T) -> Self {
+        self.request.generation = v.into();
+        self
+    }
+
+    /// Sets the value of [read_offset][control::model::ReadObjectRequest::read_offset].
+    pub fn set_read_offset<T: std::convert::Into<i64>>(mut self, v: T) -> Self {
+        self.request.read_offset = v.into();
+        self
+    }
+
+    /// Sets the value of [read_limit][control::model::ReadObjectRequest::read_limit].
+    pub fn set_read_limit<T: std::convert::Into<i64>>(mut self, v: T) -> Self {
+        self.request.read_limit = v.into();
+        self
+    }
+
+    /// Sets the value of [if_generation_match][control::model::ReadObjectRequest::if_generation_match].
+    pub fn set_if_generation_match<T>(mut self, v: T) -> Self
+    where
+        T: std::convert::Into<i64>,
+    {
+        self.request.if_generation_match = std::option::Option::Some(v.into());
+        self
+    }
+
+    /// Sets or clears the value of [if_generation_match][control::model::ReadObjectRequest::if_generation_match].
+    pub fn set_or_clear_if_generation_match<T>(mut self, v: std::option::Option<T>) -> Self
+    where
+        T: std::convert::Into<i64>,
+    {
+        self.request.if_generation_match = v.map(|x| x.into());
+        self
+    }
+
+    /// Sets the value of [if_generation_not_match][control::model::ReadObjectRequest::if_generation_not_match].
+    pub fn set_if_generation_not_match<T>(mut self, v: T) -> Self
+    where
+        T: std::convert::Into<i64>,
+    {
+        self.request.if_generation_not_match = std::option::Option::Some(v.into());
+        self
+    }
+
+    /// Sets or clears the value of [if_generation_not_match][control::model::ReadObjectRequest::if_generation_not_match].
+    pub fn set_or_clear_if_generation_not_match<T>(mut self, v: std::option::Option<T>) -> Self
+    where
+        T: std::convert::Into<i64>,
+    {
+        self.request.if_generation_not_match = v.map(|x| x.into());
+        self
+    }
+
+    /// Sets the value of [if_metageneration_match][control::model::ReadObjectRequest::if_metageneration_match].
+    pub fn set_if_metageneration_match<T>(mut self, v: T) -> Self
+    where
+        T: std::convert::Into<i64>,
+    {
+        self.request.if_metageneration_match = std::option::Option::Some(v.into());
+        self
+    }
+
+    /// Sets or clears the value of [if_metageneration_match][control::model::ReadObjectRequest::if_metageneration_match].
+    pub fn set_or_clear_if_metageneration_match<T>(mut self, v: std::option::Option<T>) -> Self
```

Comment:
In generated code we use `std::option::Option` and `std::convert::Into` to avoid conflicts, but that is not idiomatic. We should use `Option<>` and `Into<>` in hand-written code.

Reply:
Fixed in #2272